### PR TITLE
[PRM-2668] Added code for truncating file names over agreed length

### PR DESCRIPTION
**Pull request by JBlacklaws**

### CODE REVIEW

File: app/services/AppealService.scala [hunk 1]
```diff
@@ -125,4 +125,17 @@ class AppealService @Inject()(appealsConnector: PEGAConnector,
       None
     }
   }
+
+  def truncateFilename(fileName: String): String = {
+    logger.info(s"[AppealService][truncateFilename] File name length: ${fileName.length}, truncating to ${appConfig.maximumFilenameLength}")
```
Comment (LewisMitchell) [inline app/services/AppealService.scala:130]: This may need to change slightly, for: 'file1.txt' it would say: 'File name length: 9' which is wrong as the file name (without the extension) is 5

Comment (LewisMitchell) [reply app/services/AppealService.scala:130]: Also may be worth having a file reference logged so we can trace it

File: app/services/AppealService.scala [hunk 2]
```diff
@@ -125,4 +125,17 @@ class AppealService @Inject()(appealsConnector: PEGAConnector,
...
+    if(fileName.contains(".")) {
+      val fileNameSplit = fileName.split("\\.")
+      val fileNameMain = fileNameSplit.head
```
Comment (LewisMitchell) [inline app/services/AppealService.scala:133]: I haven't tested this myself but this may be an incorrect assumption. How would this react if we had a file name: 'this.is.a.file.txt'?

File: app/services/AppealService.scala [hunk 3]
```diff
@@ -125,4 +125,17 @@ class AppealService @Inject()(appealsConnector: PEGAConnector,
...
+      val fileNameExtension = "." ++ fileNameSplit.last
+
+      fileNameMain.substring(0, Math.min(fileNameMain.length(), appConfig.maximumFilenameLength - fileNameExtension.length))  ++fileNameExtension
```
Comment (LewisMitchell) [inline app/services/AppealService.scala:136]: The 150 char rule only applies to the file name and not the extension

File: test/services/AppealServiceSpec.scala [hunk 4]
```diff
@@ -127,7 +129,7 @@ class AppealServiceSpec extends SpecBase {
             fileStatus = UploadStatusEnum.READY,
             downloadUrl = Some("/"),
             uploadDetails = Some(UploadDetails(
-              fileName = "file1",
+              fileName = "file1.txt",
```
Comment (LewisMitchell) [inline test/services/AppealServiceSpec.scala:132]: Why did you need to change this?

File: app/services/AppealService.scala (hunk 3 above)
Comment (LewisMitchell) [inline app/services/AppealService.scala:136]:
```suggestion
      fileNameMain.substring(0, Math.min(fileNameMain.length(), appConfig.maximumFilenameLength - fileNameExtension.length))  ++ fileNameExtension
```

**Review by LewisMitchell — DISMISSED** [review]

File: app/services/AppealService.scala (hunk 2 above)
Comment (AndrewRButcher) [reply app/services/AppealService.scala:133]: I have seen files being uploaded where . has been used for date separators for example so this will need a re think
`VAT Return 07.04.21.pdf`

Comment (JBlacklaws) [reply app/services/AppealService.scala:133]: I could probably wrangle the regex to find the last . and split it from that

Comment (LewisMitchell) [reply app/services/AppealService.scala:133]: I think rather than doing `.head` you could drop the last element (which would be the file extension)

Comment (AndrewRButcher) [reply app/services/AppealService.scala:133]: Not sure what order this will be done but if it is before the file extension adding we will have to be careful of cases where it is missing, just something else to consider.

Comment (JBlacklaws) [reply app/services/AppealService.scala:133]: We have a list of accepted file types, right? Could we maybe we could check against their extensions?

File: test/services/AppealServiceSpec.scala [hunk 5]
```diff
@@ -146,7 +148,7 @@ class AppealServiceSpec extends SpecBase {
             informationType = "S18",
             file = SDESNotificationFile(
               recipientOrSender = "123456789012",
-              name = "file1",
+              name = "file1.txt",
```
Comment (AndrewRButcher) [inline test/services/AppealServiceSpec.scala:151]: See above
```suggestion
              name = "file1",
```

**Review by AndrewRButcher — CHANGES_REQUESTED** [review]

**Review by AndrewRButcher — APPROVED** [review]

**Comment by AndrewRButcher** [comment]
rebuild
